src/main/java/com/scaleunlimited/flinkcrawler/tools/CrawlTool.java

```diff
@@ -55,6 +57,34 @@
         private String _cacheDir;
         private String _commonCrawlId;
         
+        @Option(name = "-agent", usage = "user agent info, format:'name,email,website'", required = false)
+        public void setUserAgent(String agentNameWebsiteEmailString) {
+            if (isCommonCrawl()) {
+                throw new RuntimeException("user agent not used in common crawl mode");
+            }
```

Comment:
I think you have to do this check _after_ all options have been processed, so it's not order dependent.

Reply:
I think it's OK as long as each option checks the other one's state.